File: GridView/GridCustomHeaderCheckox/grid custom header checkbox VB/MyGridCheckBoxHeaderCellElement.vb
Imports System
Imports Telerik.WinControls
Imports Telerik.WinControls.Enumerations
Imports Telerik.WinControls.UI

Namespace _1171173
	Friend Class MyGridCheckBoxHeaderCellElement
		Inherits GridHeaderCellElement

		Private checkBox As RadCheckBoxElement
		Public Sub New(ByVal column As GridViewColumn, ByVal row As GridRowElement)
			MyBase.New(column, row)

		End Sub
		Protected Overrides Sub CreateChildElements()
			MyBase.CreateChildElements()

			checkBox = New RadCheckBoxElement()
			AddHandler checkBox.ToggleStateChanged, AddressOf CheckBox_ToggleStateChanged
			Me.Children.Add(checkBox)
		End Sub

		Private Sub CheckBox_ToggleStateChanged(ByVal sender As Object, ByVal args As StateChangedEventArgs)

			If args.ToggleState = Telerik.WinControls.Enumerations.ToggleState.On Then
				For Each item In Me.GridControl.ChildRows
					item.Cells(Me.ColumnIndex).Value = True
				Next item
			Else
				For Each item In Me.GridControl.ChildRows
					item.Cells(Me.ColumnIndex).Value = False
				Next item
			End If

		End Sub
		Public Overrides Function IsCompatible(ByVal data As GridViewColumn, ByVal context As Object) As Boolean
			Return MyBase.IsCompatible(data, context) AndAlso data.Name = "Bool"
		End Function
		Protected Overrides ReadOnly Property ThemeEffectiveType() As Type
			Get
				Return GetType(GridHeaderCellElement)
			End Get
		End Property
	End Class

	Public Class CustomColumn
		Inherits GridViewCheckBoxColumn

		Public Sub New(ByVal fieldName As String)
			MyBase.New(fieldName)
		End Sub

		Public Overrides Function GetCellType(ByVal row As GridViewRowInfo) As Type
			If TypeOf row Is GridViewTableHeaderRowInfo Then
				Return GetType(MyGridCheckBoxHeaderCellElement)
			End If
			Return MyBase.GetCellType(row)
		End Function

	End Class
End Namespace

File: GridView/GridCustomHeaderCheckox/grid custom header checkbox/MyGridCheckBoxHeaderCellElement.cs
using System;
using Telerik.WinControls;
using Telerik.WinControls.Enumerations;
using Telerik.WinControls.UI;

namespace _1171173
{
    internal class MyGridCheckBoxHeaderCellElement : GridHeaderCellElement
    {
        private RadCheckBoxElement checkBox;
        public MyGridCheckBoxHeaderCellElement(GridViewColumn column, GridRowElement row) : base(column, row)
        {

        }
        protected override void CreateChildElements()
        {
            base.CreateChildElements();

            checkBox = new RadCheckBoxElement();
            checkBox.ToggleStateChanged += CheckBox_ToggleStateChanged;
            this.Children.Add(checkBox);
        }

        private void CheckBox_ToggleStateChanged(object sender, StateChangedEventArgs args)
        {
            this.MasterTemplate.BeginUpdate();
            if (args.ToggleState == Telerik.WinControls.Enumerations.ToggleState.On)
            {
                foreach (var item in this.GridControl.ChildRows)
                {
                    item.Cells[this.ColumnIndex].Value = true;
                }
            }
            else
            {
                foreach (var item in this.GridControl.ChildRows)
                {
                    item.Cells[this.ColumnIndex].Value = false;
                }
            }
            this.MasterTemplate.EndUpdate(true, new DataViewChangedEventArgs(ViewChangedAction.DataChanged));

        }
        public override bool IsCompatible(GridViewColumn data, object context)
        {
            return base.IsCompatible(data, context) && data.Name == "Bool";
        }
        protected override Type ThemeEffectiveType
        {
            get
            {
                return typeof(GridHeaderCellElement);
            }
        }
    }

    public class CustomColumn : GridViewCheckBoxColumn
    {
        public CustomColumn(string fieldName) : base(fieldName)
        {
        }

        public override Type GetCellType(GridViewRowInfo row)
        {
            if (row is GridViewTableHeaderRowInfo)
            {
                return typeof(MyGridCheckBoxHeaderCellElement);
            }
            return base.GetCellType(row);
        }

    }
}